lessons/07-apis-debugging/README.md

```diff
@@ -160,3 +160,40 @@ app.use('/api*', function(req, res, next) {
 
 ###Can I set up my own OAuth server?
 So you want to use OAuth to authenticate and authorize your API users, huh? Unfortunately that's _just_ outside the scope of this class, but know that if you do get to that point someday, there are some great packages available (on npm) that can help you get a basic setup running fairly quickly.
+
+###Sessions and Cookies
+Now that we under stand how APIs are designed and authentication works, let's talk about a common use of external services, and that is logging in users. Currently, a large number of apps and web services allow other services to log you into their app. This is the infamous "Log in with ________" buttons on these apps login screens. When a user is logged into an app, a session is either restarted or created if it is a user's first time using the app. Sessions are just stores of information about the current user, generally done by associating a cookie with a user. Cookie's are a small amount of information sent back and forth between the server and the browser. Their are two types of cookies, but for all our purposes we will stick to session cookies, which are deleted from the browser when the page is closed (the other type are persistent, and actually our cookies will generally just contain the session id as we will use the server to hold most of the user's data). 
```

Comment:
Apps usually handle authentication by saving your information to a session, which is a store of information about the current user. The session will be restarted if it already exists or created if it's your first time using the app. A session usually works by associating a cookie with a user; a cookie is a small amount of information which is sent back and forth between the server and the browser. There are two types of cookies, but for all our purposes we will stick to session cookies, which are deleted from the browser when the page is closed (the other type is a persistent cookie, which isn't deleted from the browser when the page is closed). Usually our cookies will just contain the session id, because we will use the server to hold most of the user's data.

[MAYBE THIS IS MOVES TO THE OAUTH SECTION?]
Now that we understand how APIs are designed and how authentication works, let's talk about a common use of external services -- logging in users. You've probably seen websites with "Log in with Google" or "Log in with Facebook" buttons -- those buttons access the Google or Facebook API to sign you in with your Google or Facebook credentials.